src/main/java/bio/terra/flight/study/create/CreateStudyMetadataStep.java

```diff
@@ -24,17 +26,21 @@ public StepResult doStep(FlightContext context) {
         FlightMap inputParameters = context.getInputParameters();
         StudyRequestModel studyRequest = inputParameters.get("request", StudyRequestModel.class);
         Study newStudy = StudyJsonConversion.studyRequestToStudy(studyRequest);
-//        UUID studyid =
-        studyDAO.create(newStudy);
-        // TODO: get the id back, fetch the Study by ID and return a summary with created_date
+        studyDao.create(newStudy);
         StudySummaryModel studySummary = StudyJsonConversion.studySummaryFromStudy(newStudy);
         workingMap.put("response", studySummary);
         return StepResult.getStepResultSuccess();
     }
 
     @Override
     public StepResult undoStep(FlightContext context) {
-        // TODO: delete study with DAO
+        FlightMap inputParameters = context.getInputParameters();
+        StudyRequestModel studyRequest = inputParameters.get("request", StudyRequestModel.class);
+        String studyName = studyRequest.getName();
+        Optional<Study> study = studyDao.retrieveByName(studyName);
```

Comment:
I don't think you need to do the retrieve before the delete. And because they are in different transactions (I think?), the retrieve will not guarantee that the delete will work.

I'd suggest making delete return "found/not found" flag and ignoring the not found case in this undo. I think that's simpler.